packages/babel-plugin-minify-dead-code-elimination/__tests__/dead-code-elimination-test.js

```diff
@@ -2472,4 +2454,40 @@ describe("dce-plugin", () => {
       }
     `
   );
+
+  thePlugin(
+    "should optimize to void 0 for lets referenced before init declarations",
+    `
+      function foo() {
+        bar(a);
+        let a = 1;
+      }
+    `,
+    `
+      function foo() {
+        bar(void 0);
```

Comment:
The transformation is not a babel-bug. It's a bug in one-use-replacements in DCE. The parsing is a babel-bug.